bundler/lib/bundler/settings.rb

```diff
@@ -102,10 +102,13 @@ def initialize(root = nil)
     def [](name)
       key = key_for(name)
 
-      values = configs.values
-      values.map! {|config| config[key] }
-      values.compact!
-      value = values.first
+      value = nil
+      configs.each do |_, config|
+        if config[key]
+          value = config[key]
+          break
+        end
```

Comment:
I changed my code to this, which worked.
```suggestion
        value = config[key]
        next if value.nil?
        break
```

Reply:
Confirmed this fixes the tests, and performance is the same as it was 🎉